llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -800,11 +827,16 @@ static TableFiles processOneModule(std::unique_ptr<Module> M, bool IsEsimd,
   }
 
   {
-    ImagePropSaveInfo ImgPSInfo = {
-        true,          DoSpecConst,         SetSpecConstAtRT,
-        SpecConstsMet, EmitKernelParamInfo, EmitProgramMetadata,
-        IsEsimd};
-    string_vector Files = saveDeviceImageProperty(ResultModules, ImgPSInfo);
+    ImagePropSaveInfo ImgPSInfo = {true,
+                                   DoSpecConst,
+                                   SetSpecConstAtRT,
+                                   SpecConstsMet,
+                                   EmitKernelParamInfo,
+                                   EmitProgramMetadata,
+                                   EmitExportedSymbols,
+                                   IsEsimd};
```

Comment:
On the one hand, you can access `EmitExportedSymbols` directly within `saveDeviceImageProperty` so you don't need to modify `ImagePropSaveInfo`. But on the other hand, making a local copy of it reduces the amount of global variables used by `saveDeviceImageProperty` allowing us to refactor the origin of `EmitExportedSymbols` flag within `ImagePropSaveInfo` struct.

I'm fine with the suggested change, but it is still a bit weird to me that we wrap some data into a struct even if that data is already available in the target function where we need it.

Reply:
The formatter went a bit wild here. I don't think I have changed the behavior.

Reply:
>I don't think I have changed the behavior.

Yeah, I'm not saying that you have changed it. My point was that you could avoid extending `ImagePropSaveInfo` structure at all

Reply:
Ah, I see what you mean. I think for consistency it should do the same as the other options, but I agree it is a bit redundant.